serialization.rst

```diff
@@ -86,6 +99,13 @@ Examples
     }
   } => { 'foo' : 42, 'prot' : "wine" }
 
+  ThatArrayClass implements MongoDB\BSON\ArraySerializable {
+    private $elements => [ 1, 2, 3 ],
+    function bsonSerialize() : array {
+        return $this->elements;
+    }
+  } => [ 1, 2, 3 ]
```

Comment:
> This example conflicts with "can not be a top-level BSON document"...

I didn't think these examples were demonstrating a direct call to `fromPHP()` since they were just showing mappings, but I see your point.

I'll create three distinct examples. One with ThatArrayClass as a property within a top-level document, and two error cases (trying to serialize ThatArrayClass as a top-level document and its `bsonSerialize()` method returning an associative array).